test/boost/s3_test.cc

```diff
@@ -54,6 +55,20 @@ s3::endpoint_config_ptr make_minio_config() {
  * unrestricted anonymous read-write access
  */
 
+template<Future Fut>
+auto do_http(Fut&& fut) {
+    try {
+        if constexpr (std::is_same_v<Fut, future<>>) {
+            return fut.get();
+        } else {
+            return fut.get0();
+        }
+    } catch (httpd::unexpected_status_error& e) {
+        BOOST_FAIL(fmt::format("{}: {}", e.what(), e.status()));
```

Comment:
You can raise http log level to debug and it'll print you the _whole_ response with bad status, not only the status itself. Isn't it more helpful?

Reply:
yeah. it'd be helpful when debugging a failure which can be reproduced locally. but not for postmortem debugging. also, i am only interested in the whole reponse when something bad happens. the debug log level does not tell expected responses from unexpected ones.

Reply:
> also, i am only interested in the whole reponse when something bad happens. the debug log level does not tell expected responses from unexpected ones.

It does:

https://github.com/scylladb/seastar/blob/17183ed4e475849e5c624c876cf37f6eabc64220/src/http/client.cc#L314

Reply:
> but not for postmortem debugging. also, i am only interested in the whole reponse when something bad happens

Yes, makes perfect sense. How about equipping the s3_test with `--logger-log-level http=debug --logger-log-level s3=debug` options in the suite.yaml file? This should print much more interesting info rather than just failed response.

Reply:
> > also, i am only interested in the whole reponse when something bad happens. the debug log level does not tell expected responses from unexpected ones.
> 
> It does:
> 
> https://github.com/scylladb/seastar/blob/17183ed4e475849e5c624c876cf37f6eabc64220/src/http/client.cc#L314

thank you! i see. so it prints the whole message only when the log level is debug.

> > but not for postmortem debugging. also, i am only interested in the whole reponse when something bad happens
> 
> Yes, makes perfect sense. How about equipping the s3_test with `--logger-log-level http=debug --logger-log-level s3=debug` options in the suite.yaml file? This should print much more interesting info rather than just failed response.

this makes a lot of sense. lemme repurpose this pull request to implement this instead. that'd be more helpful. and when i debug the test manually, i will apply the same set set of options.